luigi/contrib/spark.py

```diff
@@ -298,8 +299,9 @@ def app_command(self):
         return [self.app, pickle_loc] + self.app_options()
 
     def run(self):
-        self.run_path = tempfile.mkdtemp(prefix=self.name)
-        self.run_pickle = os.path.join(self.run_path, '.'.join([self.name.replace(' ', '_'), 'pickle']))
+        name = re.sub(r'[^\w]', '_', self.name)
```

Comment:
I'd rename variable `name` to something like `run_path_prefix` (maybe some better option) due to two reasons: 
- it is actually a path, not a name anymore
- we must be having a hard time dealing with `name` and `self.name` in the same place

Reply:
Good point! Renamed.